src/ObjLoading/Parsing/LocalizeFile/Sequence/SequenceLocalizeFileLanguageValue.cpp

```diff
@@ -31,5 +31,13 @@ void SequenceLocalizeFileLanguageValue::ProcessMatch(LocalizeFileParserState* st
     state->m_current_reference_languages.emplace(langName);
 
     if (langName == state->m_language_name_caps)
-        state->m_entries.emplace_back(state->m_current_reference, valueToken.StringValue());
+    {
+        const auto& currentReference = state->m_current_reference;
+        if (const auto i = state->m_entries.find(currentReference); i != state->m_entries.end())
```

Comment:
The state is only for the current parsing session.
Meaning this works for duplicate entries in the same file.
When you have two separate str files that contain the same keys, it will not be noticed however.

My suggestion to solve this is to have a ZoneAssetLoaderState.
There is an interface called `IZoneAssetLoaderState`. If you inherit from it you can create a class that can be instanced once per zone and can hold a state across multiple asset loading calls per zone.

To retrieve it, you can do:
```
manager->GetAssetLoadingContext()->GetZoneAssetLoaderState<LocalizeZoneState>();
```
in any AssetLoader's `LoadFromRaw` class.
It will give you a state of the given class or will construct it if it was not created yet (needs to have a public default constructor).

So you could for example make a `LocalizeZoneState` class and initialize the `LocalizeFileParserState` with a pointer to it.
The zone state could hold a set of all existing keys in the zone.

Reply:
Pushed a commit to address this.